### PR TITLE
feat: support trace log level

### PR DESCRIPTION
**Pull request by hilaryRope**
## Description of the change
Add support for logtrace Go SDK

## Issue Reference
https://github.com/extism/go-sdk/issues/72 that is a logical step after [this](https://github.com/extism/go-pdk/issues/37)

## Checklist
- [] Documentation added/updated

### CODE REVIEW

File: wasm/host.wasm
Comment (G4Vi) [inline wasm/host.wasm:1]: Why regenerate this file? `host.go` doesn't look related. Was this done because the pdk was updated?

**Review by G4Vi — CHANGES_REQUESTED** [review]
Overall, looks good, thank you! I'm just unsure about why `host.wasm` being regenerated. Changes may not be required with an explanation about it.

File: wasm/host.wasm
Comment (hilaryRope) [reply wasm/host.wasm:1]: Thanks, no, I agree, this PR was still WIP, but made a mistake and did not set it to Draft. It is not needed here. I need to check the test failures as I am not quite sure why is that. I used [Modsurfer](https://modsurfer.dylibso.com/) to look into the generate `log.wasm` and I don't see the `log_trace` and I believe that is why the test is failing. Not quite sure why, though.

Comment (G4Vi) [reply wasm/host.wasm:1]: No problem, the lack of `log_trace` does appear to be at least part of the problem, it appears changes need to be made to the Go PDK to make `log_trace` available there. For example, https://github.com/extism/go-pdk/blob/main/env.go, does not import it.

Comment (hilaryRope) [reply wasm/host.wasm:1]: Indeed, I need to re-instate that first, leave it with me and will sort both. Thanks

Comment (hilaryRope) [reply wasm/host.wasm:1]: And I was the one who removed it as a temporarily fix [here](https://github.com/extism/go-pdk/issues/37) :(

File: plugins/log/go.mod
Comment (hilaryRope) [inline plugins/log/go.mod:5]: pointing to this: https://github.com/extism/go-pdk/commit/7cb3d48ce7ccff27c3eab9edb251d56caabb5c18

File: wasm/host.wasm
Comment (hilaryRope) [reply wasm/host.wasm:1]: @G4Vi changes made - I wanted to see the CICD running, but I see a message saying 1 `workflow awaiting approval`

Comment (G4Vi) [reply wasm/host.wasm:1]: Workflow approved. https://github.com/extism/go-sdk/actions/runs/10289743772/job/28485520244?pr=73

Comment (hilaryRope) [reply wasm/host.wasm:1]: @G4Vi thank you, bear with me, I will figure out what I am missing. Thanks

**Comment by nilslice** [comment]
hey just pinging this thread to see if this is blocked by anything or if I can help move it along 🙂

**Comment by G4Vi** [comment]
> hey just pinging this thread to see if this is blocked by anything or if I can help move it along 🙂

This was left on waiting for @hilaryRope to fix ci. If it isn't fixed soon we should potentially revert the pdk update.

**Comment by hilaryRope** [comment]
> > hey just pinging this thread to see if this is blocked by anything or if I can help move it along 🙂
> 
> This was left on waiting for @hilaryRope to fix ci. If it isn't fixed soon we should potentially revert the pdk update.

Hi there, sorry but I have been away. I am referencing the latest commit with my change, but I seem to be getting the same test failures. I will need to check this. Can you give me a day, please? Unless, there is something very obvious about it. Thanks

**Comment by G4Vi** [comment]
> > > hey just pinging this thread to see if this is blocked by anything or if I can help move it along 🙂
> > 
> > 
> > This was left on waiting for @hilaryRope to fix ci. If it isn't fixed soon we should potentially revert the pdk update.
> 
> Hi there, sorry but I have been away. I am referencing the latest commit with my change, but I seem to be getting the same test failures. I will need to check this. Can you give me a day, please? Unless, there is something very obvious about it. Thanks

Sure thing, thanks for the update!

**Comment by nilslice** [comment]
No rush at all :) didnt want this to get left behind in case it was waiting on something from our end.

**Comment by hilaryRope** [comment]
> No rush at all :) didnt want this to get left behind in case it was waiting on something from our end.

@nilslice  Noo, you and @G4Vi are absolutely right and it has been on the back of my mind for a bit. I am scratching my head, perhaps stupidly, as I am pointing to the correct sha in the go-pdk where the trace has been re-instante and still I am getting a failure in https://github.com/extism/go-sdk/blob/9748ea03f3dc755a575a3bdc6aa3fa7573b745ac/extism_test.go#L455. I wonder if you can have a quick check as I probably missing something trivial here. Thank you

**Comment by nilslice** [comment]
Thanks, @hilaryRope! I'll run CI and see what happens, and can investigate today too

**Comment by nilslice** [comment]
@hilaryRope @G4Vi - i'm going to merge this and make the tweaks from a new branch, as I am working on some related changes that may make this PR harder to merge later. 

The test looks ok, and I think it's just a problem finding the package at the ref. In any case, I won't tag a new release until all of this is sorted so these changes shouldn't impact anyone using the SDK.

**Comment by hilaryRope** [comment]
> @hilaryRope @G4Vi - i'm going to merge this and make the tweaks from a new branch, as I am working on some related changes that may make this PR harder to merge later.
> 
> The test looks ok, and I think it's just a problem finding the package at the ref. In any case, I won't tag a new release until all of this is sorted so these changes shouldn't impact anyone using the SDK.

Thank you for this, appreciated!

**Pull request merged**
